### ISSUE TITLE
Scheduler rework

### ISSUE THREAD

**Issue by abourget** [streamingfast/substreams#226]
A few goals:
* Have the scheduler pilot the merging, and not be at the mercy of the squasher.
* Simplifier, predictable squashing
  * Cases: support poking the storage perdiocally, if that helps. Also prepare the way for some more central job processing completion discovery (etcd?).
  * The Scheduler could interrupt the scheduling, and the squasher processing, if it discovers an already processed partial. Perhaps it scheduled a Job, but another request elsewhere (on a different pod) finished it first, and this Scheduler discovers the partial already done (either through polling the storage, or via a message from etcd or something).
* Have a more granual status of the Job, and its state, like: `Squashing`, `Pending`, `Running`, other states?
  * Goal here is to allow the Scheduler to have a periodic timer (500ms?) that can compute the stats of how many merge operations are ongoing, how many pending jobs, running jobs, predicted time to completion, etc.. And send fully computed messages that downstream sinks can simply display, instead of having all the computations to do itself (like we're doing with Progress Messages right now).
* There was an issue with the Scheduler planning CRAZY number of jobs, because someone asked for a stop block close to MaxUint32.. so we could COMPUTE the number of jobs (by divigint by the jobs segment size), but not necessarily PLAN and create those job objects, if we don't yet need to.
  * The algorithm could be more directly like the diagonal of being one module ahead for the prior stage.
* Allow the Scheduler to dynamically adjust the number of parallel jobs, so we can start with 1 or 2, and scale up or down depending on some perf metrics or progress whatever.
  * At the same time, if it can "produce" the next pending jobs ad-hoc, then it keeps everything very lean.
* The Scheduler has the opportunity to become much simplified, if we treat it as a matrix of stages + windowed ranges, instead of a complex graph of modules with complex dependencies.
  * We can schedule a stage to be executed all at once: 1) ideally, we could schedule a SINGLE Tier2 request, with all of the modules of the given stage, but to go faster, we will: 2) schedule a stage by scheduling multiple Tier2 jobs in a `llerrgroup`, and return the stage execution for all modules within that stage, only when all those jobs are done properly. All of those modules' storages will then be merged in parallel.

**Comment by abourget** [comment]
I imagine the pattern of `bubbletea` for the runloop of the Scheduler, where inputs arrive from different places, all decision making is done synchronously in that loop, so no locks for updating the workPlan, computing all the stats, etc.. and multiple feeders (for instancne, of partials presence) can send the same signal into the loop, triggering the same effects (kill the currently running job, start the merging operation, da da da ).

**Comment by abourget** [comment]
Started in branch feature/scheduler-rework

**Comment by sduchesneau** [comment]
Merged in develop, deployed.
